enhancements/etcd/dr-leveraging-ceo-scaling.md

```diff
@@ -0,0 +1,151 @@
+```
+2. Starting the etcd pod should be started with the same value of $ETCD_NAME, ETCD_INITIAL_CLUSTER and $ETCD_NODE_PEER_URL as in step 1.
+
+This will allow etcd to populate the data directory with a new cluster ID, and new membership data that includes only 1 member.
+
+Restoring a single member from a backup and scaling up has two distinct problem, they are:
+1. Get the first etcd member running. 
+2. Get the operator to deploy static pods that can decide that they are being restored, so they dont start on the existing data directories.
+
+### Static Pod identifying a restore
+
+Assuming the first member is up, the control plane will be up and operator start running. A user can trigger a new rollout of the etcd pods by running `oc patch forceredpeloymment=reason-time`. New pods will be rolled out on the remaining masters that have to be restored. Before etcd commands on these pods are run, the existing data directory needs to be removed, because they contain the previous cluster state information.
+
+One possible solution is, during the start up pod queries etcd membership, if the pod is an unstarted member, it can be assumed that the data directory present is from previous cluster and moved out.
+
+Other solution is to check the cluster id of the new cluster with the cluster id of the existing data directory, move out the data directory only if they are different. This is a little time consuming to implement and test.
+
+### Getting the first etcd member running
+
+This will involve bringing a control plane up with a single node etcd cluster and then scaling it up to three, the steps are:
+  1. move the existing etcd pod yaml out of kubelet manifest directory and move etcd data directory into a different location.
+  2. copy the snapshot file from backup directory to /var/lib/etcd-backup/snapshot.db
+  3. move the kube-apiserver backed up revision back into kubelet manifest directory and delete other revisions and related resources(certs and configmaps).
+  4. copy the restore-etcd-pod.yaml from backed up resources to /etc/kubernetes/manifests/etcd-pod.yaml. This will restore a single member control plane using the backed up snapshot.db and then start the etcd pod bringing the control plane up. 
+  5. restart the kubelet.
+  6. force CEO redeployment via `oc patch forceredpeloymment=reason-time`
+  7. When CEO redeployment is forced, the existing nodes will start with new pods, a scale up similar to bootstrap will take place
+  8. CEO will redeploy the modified pod on recovery master node to the current revision.
+
+## Design Details
+
+TODO
+
+### Test Plan
+
+TODO
+
+### Graduation Criteria
+
+TODO
+
+##### This feature will GA in 4.3 as long as:
+
+TODO
+
+### Upgrade / Downgrade Strategy
+
+TODO
+
+#### Upgrade from 4.3 to 4.4
+
+TODO
+
+#### Downgrade from 4.4 to 4.3
+
+TODO
+
+### Version Skew Strategy
+
+TODO
+
+## Implementation History
+
+TODO
+
+## Drawbacks
+
+TODO
```

Comment:
clearly explain what happens if I have state like this:
1. master1 has the restore-etcd-pod running
2. master2 and master3 are actually in a quorum together
3. master2 gets restarted with the current static pod

If we can make this scenario function correctly, then we can simplify further to having the presence of a backup and absence of a data-dir mean, "restore from this backup".

Reply:
added